### PR TITLE
Add complete multi-fov ISS vignette

### PR DESCRIPTION
**Pull request by ambrosejcarr**
Adds an ISS Vignette. Not yet ready for review. 

- [x] Add vignette
- [ ] discuss and implement testing strategy for contained code
- [x] rebase to harmonize with API changes needed to support this PR. 

Testing strategy: 
```
make docs-html
open docs/build/html/index.html
```
navigate to Starfish Usage -> In-Situ Sequencing & Read!

### CODE REVIEW

File: docs/source/usage/iss/iss_vignette.rst [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+experiments can be downloaded on-demand into starfish. However, downloading the raw data requires
+that you access the data from an aws account.
+
+For the purposes of this vignette, we will format only two of the 16 fields of view. To download the
+data, you can run the following commands:
+
+.. code-block:: bash
+
+    mkdir -p iss/raw
+    aws s3 cp s3://czi.starfish.data.public/browse/raw/20180820/iss_breast/ iss/raw/ \
```
Comment (joshmoore) [inline docs/source/usage/iss/iss_vignette.rst]: This leads to `fatal error: Unable to locate credentials`. Workaround for the general public is to add `--no-sign-request`. Tested via:

```
FROM alpine:3.6
RUN apk -v --update add \
        python \
        py-pip \
        groff \
        less \
        mailcap \
        && \
    pip install --upgrade awscli==1.14.5 s3cmd==2.0.1 python-magic && \
    apk -v --purge del py-pip && \
    rm /var/cache/apk/*
WORKDIR /data
ENTRYPOINT ["aws"]
```

Comment (ambrosejcarr) [reply docs/source/usage/iss/iss_vignette.rst]: Awesome, is that a general work-around for use of links with public permissions?

Comment (joshmoore) [reply docs/source/usage/iss/iss_vignette.rst]: `aws s3 --no-sign-request ...` has worked for every read operation I've tried so far (e.g. `cp`, `ls`, `sync`)

**Comment by ambrosejcarr** [comment]
Merging this per our discussion during the Eng-sync. We will add it to master, and then @shanaxel42 will take a look at adding tests for the files that it includes.

File: docs/source/usage/iss/iss_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+def iss_pipeline(fov, codebook):
+    primary_image = fov.primary_image
+
+    # register the raw images
+    registration = Registration.FourierShiftRegistration(
+        upsampling=1000,
+        reference_stack=fov['dots']
+    )
+    registered = registration.run(primary_image, in_place=False)
```
Comment (ttung) [inline docs/source/usage/iss/iss_pipeline.py]: in_place defaults to False now.  We should remove this.

File: docs/source/usage/iss/iss_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+    # filter raw data
+    masking_radius = 15
+    filt = Filter.WhiteTophat(masking_radius, is_volume=False)
+    filtered = filt.run(registered, verbose=True, in_place=False)
```
Comment (ttung) [inline docs/source/usage/iss/iss_pipeline.py]: ditto

File: docs/source/usage/iss/iss_pipeline.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    # detect spots using laplacian of gaussians approach
+    p = SpotFinder.GaussianSpotDetector(
+        min_sigma=1,
+        max_sigma=10,
+        num_sigma=30,
+        threshold=0.01,
+        measurement_type='mean',
+    )
+    blobs_image = fov['dots'].max_proj(Indices.ROUND, Indices.Z)
+    intensities = p.run(filtered, blobs_image=blobs_image)
```
Comment (ttung) [inline docs/source/usage/iss/iss_pipeline.py]: s/intensities/intensity_tables/

File: docs/source/usage/iss/iss_pipeline.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    assigned = ta.run(decoded, regions)
+
+    return assigned, regions
+
+
+# process all the fields of view, not just one
+def process_experiment(experiment: starfish.Experiment):
+    decoded_intensities = {}
+    regions = {}
+    for name_, fov in experiment.items():
```
Comment (ttung) [inline docs/source/usage/iss/iss_pipeline.py]: `name` is not a special kw.  would just rename it `name`.  `fov_name` would be even better.

File: docs/source/usage/iss/iss_vignette.rst [hunk 6]
```diff
@@ -0,0 +1,164 @@
+In-Situ Sequencing
+==================
+
+This in-situ sequencing (ISS) experiment has 16 fields of view that measure 4 channels on 4 separate
+imaging rounds.
+
+Downloading Data
+----------------
+
+Like all starfish data, this experiment is hosted on amazon web services. Once formatted,
```
Comment (ttung) [inline docs/source/usage/iss/iss_vignette.rst]: this experiment's data is hosted on Amazon's S3 service.

File: examples/get_iss_breast_data.py [hunk 7]
```diff
@@ -12,16 +12,14 @@
 from starfish.types import Coordinates, Indices, Number
 from starfish.util.argparse import FsExistsType
 
-SHAPE = 1044, 1390
```
Comment (ttung) [inline examples/get_iss_breast_data.py]: This was here to avoid DRY.

Comment (ambrosejcarr) [reply examples/get_iss_breast_data.py]: Had to take it out because I couldn't find a way to include a constant.

**Pull request merged**